canonical/maas-ui
# feat(machines): add support for action filters

## Done

- Add filters to the action handlers.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Open [SetPoolForm.tsx](https://github.com/canonical/maas-ui/blob/main/src/app/machines/components/MachineHeaderForms/ActionFormWrapper/SetPoolForm/SetPoolForm.tsx#L81-L90) and replace lines 82-89 with:
```
            dispatch(
              machineActions.setPool({
                pool_id: pool.id,
                filter: {
                  hostname: machines.map(({ hostname }) => hostname),
                },
              })
            );
```
- Now go to the machine list.
- Tick some machines and open the set pool form via the take action menu.
- Choose a pool and submit.
- The machines should get updated with the new pool (but the form won't close).
- In your websocket messages there should be a machine.action message that passes `filter:` instead of a number of messages that pass `system_id`.
## Fixes

Fixes: canonical/app-tribe#1137.
